lua/toggleterm.lua

```diff
@@ -338,7 +357,17 @@ function M.toggle(count, size, dir, direction)
   if count >= 1 then
     toggle_nth_term(count, size, dir, direction)
   else
-    smart_toggle(count, size, dir, direction)
+    local toggle_strategy
+    if type(config.toggle_strategy) == "function" then
+      toggle_strategy = config.toggle_strategy
```

Comment:
I think before anything is exposed this way, i.e. we take a function it needs to be ironed out how this function will work so the api can remain consistent over time. Are there existing ideas of how a user will add this or is the general idea someone will think of something in which case I'd rather wait till there is a specific use case and cater to those rather than taking a function without a sense of if it's desirable by users and what such a function would take as arguments if any.

Reply:
Sounds good, I agree let's implement strategies on our own as requests come in. Removed.